### DISCUSSION TITLE
Get selected DetailsList rows

### DISCUSSION THREAD

**Discussion by lukebandy** [Appsilon/shiny.fluent#70] [Q&A]
I imagine this is possible using JavaScript, but is there a clean way of getting this directly through R as with DT through `input$tableId_rows_selected`?

**Comment by kamilzyla** [comment]
Currently the only way is via JS. Thanks for asking though - sounds like a useful feature! I created #81 to track it.

**Comment by zilch42** [comment]
Hi @kamilzyla, could you by any chance provide an example of how to do this with JS? there isn't any onClick callback so I'm not sure how to go about it.

**Reply by kamilzyla** [reply]
After taking a look at the official docs, it seems to be quite a complex task - I'll try to come back with an example next week.

**Reply by zilch42** [reply]
Thanks I really appreciate it!

**Comment by kamilzyla** [comment]
@lukebandy @zilch42 Try this:
```r

library(shiny)
library(shiny.fluent)

CustomComponents <- tags$script(HTML("(function() {
  const React = jsmodule['react'];
  const Fluent = jsmodule['@fluentui/react'];
  const Shiny = jsmodule['@/shiny'];
  const CustomComponents = jsmodule['CustomComponents'] = {};

  function useSelection(inputId) {
    const selection = React.useRef(new Fluent.Selection({
      onSelectionChanged() {
        const value = this.getSelectedIndices().map(i => i + 1); // R uses 1-based indexing.
        Shiny.setInputValue(inputId, value);
      }
    }));
    return selection.current;
  }

  CustomComponents.DetailsList = function DetailsList({ inputId, ...rest }) {
    const selection = useSelection(inputId);
    return React.createElement(Fluent.DetailsList, { selection, ...rest });
  }
})()"))

DetailsList.shinyInput <- function(inputId, ...) {
  shiny.react::reactElement(
    module = "CustomComponents",
    name = "DetailsList",
    props = shiny.react::asProps(inputId = inputId, ...),
    deps = shinyFluentDependency()
  )
}

items <- list(
  list(name = "Apple"),
  list(name = "Banana"),
  list(name = "Cherry")
)

shinyApp(
  ui = tagList(
    CustomComponents,
    DetailsList.shinyInput("selection", items = items),
    textOutput("selection")
  ),
  server = function(input, output) {
    output$selection <- renderText(paste(input$selection, collapse = ", "))
  }
)
```

This solution is not ideal (for example, you cannot set the selection from Shiny), but it should get you started :slightly_smiling_face:

**Reply by zilch42** [reply]
Thanks! Enough for what I need at the moment. Appreciate your help

**Reply by kyle-power** [reply]
Thank you, @kamilzyla !! This custom component is incredible useful - maybe you could add it to the package as a regular component in the future?

**Reply by kamilzyla** [reply]
@kyle-power Glad to hear that :smile: Yes, we already have an [issue](https://github.com/Appsilon/shiny.fluent/issues/81) to add `DetailsList.shinyInput`.
